docs/operations/export_api.md

```diff
@@ -0,0 +1,425 @@
+# Export API
+
+## Prerequesites
+None
+
+### Export Feature Gate
+
+VMExport support must be enabled in the feature gates to be supported. The
+[feature gates](./activating_feature_gates.md#how-to-activate-a-feature-gate)
+field in the KubeVirt CR must be expanded by adding the `VMExport` to it.
+
+### Export token
+
+In order to securely export a Virtual Machine Disk, you must create a token that is used to authorize users accessing the export endpoint. This token must be in a secret that is stored in the namespace the Virtual Machine resides in. The contents of the secret can be passed as a token header or parameter to the export URL. The name of the header or argument is `x-kubevirt-export-token` with a value that matches the content of the secret. The secret can be named any valid secret in the namespace. We recommend you generate an alpha numeric token of at least 12 characters. The data key should be `token`. For example:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: example-token
+data:
+  token: MTIzNDU2Nzg5MGFi
+```
+### Export Virtual Machine volumes
+After you have created the token you can now create a VMExport CR that identifies the Virtual Machine you want to export. You can create a VMExport that looks like this:
+
+```yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+  name: example-export
+spec:
+  tokenSecretRef: example-token
+  source:
+    apiGroup: "kubevirt.io"
+    kind: VirtualMachine
+    name: example-vm
+```
+
+The following volumes present in the VM will be exported:
+
+* PersistentVolumeClaims
+* DataVolumes
+* MemoryDump
+
+All other volume types are not exported. The export will only be available when the VM is not running, this is to prevent corrupt data from being exported. If you want to export data from a running VM, the VM will have to use storage that can be snapshotted, and you can create a Virtual Machine Snapshot and export that while the VM stays running. If an export is active and the VM is started the export will detect this state and terminate the export server.
```

Comment:
To avoid the export of inconsistent data, a Virtual Machine can only be exported while it is powered off.  Any active VM exports will be terminated if the Virtual Machine is started.  To export data from a running Virtual Machine you must first create a Virtual Machine Snapshot (see below).

Reply:
Done